### PR TITLE
doc: migration from v2 to v3 📚

### PR DESCRIPTION
**Pull request by gregorylegarec**

### CODE REVIEW

**Comment by gregorylegarec** [comment]
[A trier Doc de migration de cozy-konnector-libs v2 à cozy-konnector-libs v3](https://trello.com/c/tk6QfULw/102-a-trier-doc-de-migration-de-cozy-konnector-libs-v2-%C3%A0-cozy-konnector-libs-v3)

File: docs/migrations/v2-to-v3.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Oviously, start by upgrading the `cozy-konnector-libs` package.
+
+```bash
+$> yarn upgrade cozy-konnector-libs --save
+```
+
+## Deprecated attributes
+Following konnector's attributes may be removed, as they are not used anymore:
+* `color`
```
Comment (doubleface) [inline docs/migrations/v2-to-v3.md]: And `name`, `models`, `category`, `dataType`, `customView`

File: docs/migrations/v2-to-v3.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Removed models
+The following models has been removed and code using them has to be adapted:
+* Bill
+
+## Removed methods
+The following methods has been removed from codebase:
+* `filterExisting`
+* `linkBankOperation`
+These both ones have been directly encapsulated into `saveBills`
```
Comment (doubleface) [inline docs/migrations/v2-to-v3.md]: For connectors not directly related to bills, there are saveFiles`, `filterData`, `addData` and `linkBankOperation` which are called by saveBills internally.

File: docs/migrations/v2-to-v3.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+### No more `request`, now use `request`
+The `request` method is now provided directly by `cozy-konnector-libs`, so the following import
+```js
+const request = require('request')
+```
+have to be replaced by
+```js
+const { request } = require('cozy-konnector-libs')
+```
+`request` is a instance of [`request-promise`](https://github.com/request/request-promise) with a custom configuration (`followAllRedirects` is set to `true`).
```
Comment (doubleface) [inline docs/migrations/v2-to-v3.md]: Actually, here, request is a function returning a request-promise instance with "better" defaults for connectors, but I see that the naming is misleading. Maybe requestFactory ?

**Comment by nono** [comment]
Ping @gregorylegarec

File: docs/migrations/v2-to-v3.md (hunk 3 above)
Comment (gregorylegarec) [reply docs/migrations/v2-to-v3.md]: Yes maybe a more explicit name could be a good idea, let's keep this change for later, if you're ok.

Comment (doubleface) [reply docs/migrations/v2-to-v3.md]: @gregorylegarec But then, could you replace this sentence with something like

```
`request` is a function returning an instance of [`request-promise`](https://github.com/request/request-promise) with a custom configuration (`followAllRedirects` is set to `true`).
```

**Pull request merged**